velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -209,4 +209,93 @@ struct LastDayFunction {
   }
 };
 
+template <typename T>
+struct MonthsBetweenFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Timestamp>* /*timestamp1*/,
+      const arg_type<Timestamp>* /*timestamp2*/,
+      const bool /*roundOff*/) {
+    timeZone_ = getTimeZoneFromConfig(config);
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getYear(const std::tm& time) {
+    return 1900 + time.tm_year;
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getMonth(const std::tm& time) {
+    return 1 + time.tm_mon;
+  }
+
+  FOLLY_ALWAYS_INLINE int64_t getDay(const std::tm& time) {
+    return time.tm_mday;
+  }
+
+  FOLLY_ALWAYS_INLINE std::tuple<int32_t, int32_t, int32_t, int32_t> splitDate(
+      Timestamp timestamp) {
+    std::tuple<int32_t, int32_t, int32_t, int32_t> result;
+    int32_t year = getYear(getDateTime(timestamp, this->timeZone_));
+    int32_t monthInYear = getMonth(getDateTime(timestamp, this->timeZone_));
+    int32_t dayInMonth = getDay(getDateTime(timestamp, this->timeZone_));
+    int32_t daysToMonthEnd = util::getMaxDayOfMonth(year, monthInYear) -
```

Comment:
Maybe remove these variables and create the tuple directly because they are used only once.

Reply:
done